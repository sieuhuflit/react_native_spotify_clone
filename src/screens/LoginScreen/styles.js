import { StyleSheet } from 'react-native';
import { colors } from '../../constants';

const styles = StyleSheet.create({
  logoWrapper: {
    flex: 0.8,
    justifyContent: 'center',
    alignItems: 'center',
    flexDirection: 'row',
  },
  logo: {
    alignSelf: 'center',
    height: 80,
    width: 80,
    marginRight: 20,
  },
  logoText: {
    alignSelf: 'center',
    color: 'white',
    fontSize: 45,
    fontWeight: 'bold',
  },
  loginButtonWrapper: {
    flex: 0.2,
    alignItems: 'center',
    height: 50,
    shadowColor: colors.blackBg,
    shadowOffset: { height: -10, width: 0 },
    shadowOpacity: 0.2,
    shadowRadius: 20,
  },
  loginButton: {
    backgroundColor: colors.brandPrimary,
    borderRadius: 25,
    height: 50,
    width: 220,
  },
  loginButtonText: {
    fontSize: 16,
    fontWeight: 'bold',
    color: colors.white,
    letterSpacing: 1,
    textTransform: 'uppercase',
  },
});

export default styles;
