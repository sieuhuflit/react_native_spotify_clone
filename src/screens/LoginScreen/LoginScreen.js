import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Video from 'react-native-video';
import { StyleSheet, Image, Text, View } from 'react-native';
import { globalStyles } from '../../constants';
import RoomiesVideo from '../../assets/roomies.mp4';
import TouchText from '../../components/TouchText';
import styles from './styles';

export default class LoginScreen extends Component {
  static propTypes = {
    getToken: PropTypes.func.isRequired,
    navigation: PropTypes.shape({
      navigate: PropTypes.func.isRequired,
    }).isRequired,
  };

  onPressLogin = () => {
    const { getToken, navigation } = this.props;
    getToken(() => {
      navigation.navigate('TabNavigation');
    });
  };

  render() {
    return (
      <View style={globalStyles.flex1}>
        <Video
          repeat
          source={RoomiesVideo}
          resizeMode="cover"
          style={StyleSheet.absoluteFill}
          shouldPlay
          isLooping
        />

        <View style={styles.logoWrapper}>
          <Image source={require('../../assets/logo.png')} style={styles.logo} />
          <Text style={styles.logoText}>Spotify</Text>
        </View>
        <View style={styles.loginButtonWrapper}>
          <TouchText
            onPress={this.onPressLogin}
            style={styles.loginButton}
            styleText={styles.loginButtonText}
            text="Login"
          />
        </View>
      </View>
    );
  }
}
