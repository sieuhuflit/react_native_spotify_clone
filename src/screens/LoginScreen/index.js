import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import LoginScreen from './LoginScreen';
import { getToken } from '../../data/auth/actions';

const mapStateToProps = () => ({});

const mapDispatchToProps = dispatch =>
  bindActionCreators({ getToken }, dispatch);

export default connect(mapStateToProps, mapDispatchToProps)(LoginScreen);
