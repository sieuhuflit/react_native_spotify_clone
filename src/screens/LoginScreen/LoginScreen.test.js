import React from 'react';
import { shallow } from 'enzyme';
import LoginScreen from './LoginScreen';

const setup = propOverrides => {
  const props = Object.assign(
    {
      getToken: jest.fn(),
      navigation: {
        navigate: jest.fn()
      }
    },
    propOverrides
  );
  const wrapper = shallow(<LoginScreen {...props} />);
  return {
    props,
    wrapper
  };
};

describe('LoginScreen check snapshot', () => {
  test('should match snapshot', () => {
    const { wrapper } = setup();
    expect(wrapper).toMatchSnapshot();
  });
});

describe('LoginScreen check simulate click', () => {
  test('should onPressLogin work', () => {
    const { wrapper, props } = setup();
    wrapper
      .find('TouchText')
      .props()
      .onPress();
    expect(props.getToken).toBeCalledTimes(1);
  });
});
