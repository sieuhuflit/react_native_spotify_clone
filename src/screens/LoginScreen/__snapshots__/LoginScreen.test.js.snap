// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginScreen check snapshot should match snapshot 1`] = `ShallowWrapper {}`;
